src/seraphis_main/enote_record_utils_legacy.cpp

```diff
@@ -366,6 +367,197 @@ bool try_get_legacy_basic_enote_record(const LegacyEnoteVariant &enote,
+static bool try_out_to_legacy_enote_v1(const cryptonote::transaction &tx,
+    const size_t output_index,
+    sp::LegacyEnoteVariant &enote_out)
+{
+    if (output_index >= tx.vout.size())
+        return false;
+    if (!is_legacy_enote_v1(tx, tx.vout[output_index]))
+        return false;
+
+    sp::LegacyEnoteV1 enote_v1;
+
+    /// Ko
+    crypto::public_key out_pub_key;
+    cryptonote::get_output_public_key(tx.vout[output_index], out_pub_key);
+    enote_v1.onetime_address = rct::pk2rct(out_pub_key);
+    /// a
+    enote_v1.amount = tx.vout[output_index].amount;
+
+    enote_out = std::move(enote_v1);
+    return true;
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool try_out_to_legacy_enote_v2(const cryptonote::transaction &tx,
+    const size_t output_index,
+    sp::LegacyEnoteVariant &enote_out)
+{
+    if (output_index >= tx.vout.size())
+        return false;
+     if (!is_legacy_enote_v2(tx, tx.vout[output_index]))
+        return false;
+    if (output_index >= tx.rct_signatures.outPk.size() || output_index >= tx.rct_signatures.ecdhInfo.size())
+        return false;
+
+    sp::LegacyEnoteV2 enote_v2;
+
+    /// Ko
+    crypto::public_key out_pub_key;
+    cryptonote::get_output_public_key(tx.vout[output_index], out_pub_key);
+    enote_v2.onetime_address = rct::pk2rct(out_pub_key);
+    /// C
+    enote_v2.amount_commitment = tx.rct_signatures.outPk[output_index].mask;
+    /// enc(x)
+    enote_v2.encoded_amount_blinding_factor = tx.rct_signatures.ecdhInfo[output_index].mask;
+    /// enc(a)
+    enote_v2.encoded_amount = tx.rct_signatures.ecdhInfo[output_index].amount;
+
+    enote_out = std::move(enote_v2);
+    return true;
+}
+//-------------------------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------------------------
+static bool try_out_to_legacy_enote_v3(const cryptonote::transaction &tx,
+    const size_t output_index,
+    sp::LegacyEnoteVariant &enote_out)
+{
+    if (output_index >= tx.vout.size())
+        return false;
+    if (!is_legacy_enote_v3(tx, tx.vout[output_index]))
+        return false;
+    if (output_index >= tx.rct_signatures.outPk.size() || output_index >= tx.rct_signatures.ecdhInfo.size())
+        return false;
+
+    sp::LegacyEnoteV3 enote_v3;
+
+    /// Ko
+    crypto::public_key out_pub_key;
+    cryptonote::get_output_public_key(tx.vout[output_index], out_pub_key);
+    enote_v3.onetime_address = rct::pk2rct(out_pub_key);
+    /// C
+    enote_v3.amount_commitment = tx.rct_signatures.outPk[output_index].mask;
+    /// enc(a)
+    const size_t byte_size = sizeof(enote_v3.encoded_amount);
```

Comment:
```suggestion
    constexpr size_t byte_size = sizeof(enote_v3.encoded_amount);
```

Should technically be `constexpr` to be used portably in a `static_assert` expression.